### DISCUSSION TITLE
alphatex show drum (notation or tabs)

### DISCUSSION THREAD

**Discussion by sugizo** [CoderLine/alphaTab#474] [Q&A] [answered]
# Description
with guitar pro file, alphatab can show drum notation, it'll nice to have if alphatex can show drum notation too

# Possible Solutions
```
.
\track "Drums"
\instrument 119 \clef Neutral
:4 f4{dy f} c5 f4 c5
```

**result**
not showing the cymbal notation **x|** above the notation

trying to add  **\tuning drum** on alphatex above

**result**
the notation not shown

**tested for both **score** and **tab** stave profile on **
http://sugizo.pythonanywhere.com/music/notation/form_tablature_in_notation

**n.b.**
\instrument 119 stand for Synth Drum
**ref**
https://en.wikipedia.org/wiki/General_MIDI#Percussive

# Question
is alphatex support drum notation ?
**e.g.** 
https://www.alphatab.net/docs/showcase/special-tracks/#drum-tabs 

or even better drum tabs ?
**e.g.**
https://en.wikipedia.org/wiki/Drum_tablature

```
  HH|x-x-x-x-x-x-x-x-||
   S|----o-------o---||
   B|o-------o-------||
     1 + 2 + 3 + 4 +
```

thanks

**Comment by Danielku15** [comment, marked as answer]
> is alphatex support drum notation ?

As of today alphaTex does not support any kind of drum notation. It should be fairly easy to add it since the latest changes I made for supporting percussion notation. I would need to think of a proper way to address the indivdiual instruments as the midi number is not really nice: 
https://github.com/CoderLine/alphaTab/blob/b31f6a47d24f32231110e651de4e796445b5aae2/src/model/PercussionMapper.ts#L76

I will add a work item for that to the backlog but I think it will have rather low priority. 

> or even better drum tabs ?

This would then be support for a full separate file format. and usually ASCII tabs are not really following a common standard. They are not really suited for machine reading and many people usually write it slightly adjusted which makes it impossible to properly support them. So I guess I will rather not add support for ascii drum tabs but I'm happy to accept contributions. 

>  \instrument 119 stand for Synth Drum

Midi instrument 119 might be 'Synth **Drum**' from the sound but it is not a **midi percussion** instrument. it has the same sounds for all note heights. You have to consider https://en.wikipedia.org/wiki/General_MIDI#Percussion for that.

**Comment by Danielku15** [comment]
[1.4.0-alpha.1026](https://www.npmjs.com/package/@coderline/alphatab/v/1.4.0-alpha.1026) has now support for percussion tracks via alphaTex: 

https://next.alphatab.net/docs/alphatex/percussion

**Comment by tbreuss** [comment]
I see that the question has been answered, but it is still not clear to me from the answer whether this feature has been released.

I'm wondering if it's possible to create drum patterns with AlphaTex version 1.8.0. Something like this:
https://alphatab.net/docs/showcase/special-tracks#drum-tabs

If I understand the website code correctly, these notations are created from Guitar Pro files.

@Danielku15 Sorry for the necroposting, but I didn't want to start a new discussion.

**Reply by leocaseiro** [reply]
You can.

alphatex https://music-notation-two.vercel.app/# for alphatab

Eg:

\track "Drums"\
\instrument percussion\
\tempo 120\
\clef neutral\
\articulation defaults\
\ts 4 4\
(42 36).8 (42 38).8 (42 36).8 (42 36).16 (42 36).16

**Reply by tbreuss** [reply]
Nice 👍

**Reply by leocaseiro** [reply]
I found the docs here https://alphatab.net/docs/alphatex/staff-metadata#articulation

**Reply by tbreuss** [reply]
> alphatex https://music-notation-two.vercel.app/# for alphatab

Thanks, very useful link. 

By the way, this could/should be a feature of the official AlphaTab website.
I left this in a comment on the website project:
https://github.com/CoderLine/alphaTabWebsite/issues/131
